### ISSUE TITLE
problem with VoiceOver reading information in files list

### ISSUE THREAD

**Issue by alexwallis** [litchie/ge-public#70]
I have noticed an issue where VoiceOver does strange things when navigating the files and folders list in golden ear.
The problem is this, as soon as you open a folder in golden ear depending on the item count inside a folder VoiceOver can start to read you the name of the very last item in the folder then iinterrupt itself a second into reading and then read the navigation information for the back button.
Sometimes focus is placed on the last item in the folder.
I have found that the length of file or folder names affects when this behaviour occurs.
For folders which are named with one character the behaviour triggers once you have 7 items or more inside a folder named liked this.
for example I have a folder called target, and then inside that I have folders named 1 2 3 4 5 6 and 7.
For longer folder names for example CD 05 you must have 8 items or more inside a folder named like this, if you only had 7 folders named like CD 05 then it wouldn't trigger the behaviour.
Its a minor VoiceOver issue but an annoying one particularly when focus lands on the last item in the folder instead of staying at the top of the list where it should be, and it shouldn't be trying to read the name of the last item in the list as you open the folder.
This also happens in file lists when you have folders which only have files in.
I believe if this issue can be fixed it might also address a similar issue I have noticed in some dialogues in golden ear I will find the issue numbers and add them as a comment because I haven't worked out how to edit this main description.
I have checked and I can reproduce the issue consistently.

**Comment by alexwallis** [comment]
I believe the behaviour documented in issue #44 is an example of the same bug but in dialogues. The general behaviour described in issue #44 is the same as in this.

**Comment by litchie** [comment]
Thanks. Will investigate this issue. What you want is that the first item should be read instead of the last. Correct?

**Comment by alexwallis** [comment]
As you go into folders VoiceOver tries to read the name of the last item, so for example if the last item was called cd9 VoiceOver would say c, then instantly iinterrupt itself and say back button.
The issue happens for a split second I am not sure if you will notice it,
what I want though is for the conditions to be the same whenever I enter a folder no matter the item count so the first thing read will be the back button and this issue of VoiceOver starting to read the last item then iinterrupting itself doesn't happen.
I thought the issue was dependent on item count but I have just been experimenting and I am no longer sure if this is the case.
I wonder if it could be something to do with the screen refreshing itself as I access folders.
I have just been into folders that didn't meet the conditions I thought I had identified and experienced the issue where the VoiceOver was doing this again, so saying the first letter of the last item in the folder and sometimes landing focus on it.
I hope I am explaining the issue properly if I am using complicated English or can explain it simpler please let me know.

**Comment by litchie** [comment]
Just did a quick check, here is what happens:
1. It says the folder name which is on the title bar;
2. It says "back back button".

None of the list item is spoken. This is a folder of about ten files.

**Comment by alexwallis** [comment]
I see, what you describe sounds what I would normally see happening, the item counts I mentioned above depending on the length of the folder names for me seem consistent ways to get the issue to reproduce,
though I have scene it happen outside of those conditions, would it be helpful for me to upload some example folder structures without files?
I could try to post a recording of the issue if you would find this at all helpful.
In case its of any use I have observed this issue at least 1 level in from the golden ear documents folder.
I am using the app on a 6+.

**Comment by litchie** [comment]
Yes, a folder structure and a recording will definitely help. By the way, you can just use some fake files, so that we can reproduce the problem exactly using this folder structure.

**Comment by alexwallis** [comment]
Hi, I have made a recording demonstrating the problem,
I hope this is OK, I couldn't upload the file directly so I have pasted the dropbox link below,
I should also say this issue didn't happen on the old golden ear in case that is important.
The download link for the recording is
https://dl.dropboxusercontent.com/u/12242051/Rec_7%20Jan%202015%203_47_15%20pm.m4a
I will try to get an example folder structure uploaded to dropbox for you as well, sorry about the recording quality, you might find it useful to listen on headphones, but I think the issue should be clear if you listen carefully, the speed of VoiceOver doesn't affect if this issue happens.

**Comment by litchie** [comment]
The recording quality is very good. I hear the "s" before "Target" clearly. I will try your example folder on my device and see if this problem can be reproduced.  The male voice is different from the voice used in my device, I am not sure if that could be a problem. Is it possible to try a new voice and see if this problem still exists?

**Comment by alexwallis** [comment]
Did you hear the other characters in other parts of the recording? as to the voice, the voice is provided buy apple, you get the mail voice if you go into general, accessibility, VoiceOver, speech then set the default language dialect to UK English,
I am happy to try another voice but this won't be the cause of the problem, as it didn't happen in the old version of golden ear before the big update, and Apple hasn't changed its UK Voice for years. I am using the UK English voice enhanced quality.

**Comment by alexwallis** [comment]
I just did a test using the female US English voice and the issue is still present, so clearly its not specifically the UK English voice causing the problem.

**Comment by litchie** [comment]
I find the similar problem in other apps as well. Strongly suspect that this is an OS problem.

**Comment by alexwallis** [comment]
That's interesting you think the issue could be an OS bug, what other apps have you found this happening in? I am not sure if it is an OS bug or not, as for example I haven't scene issue #44 happen in other apps and this is similar. Also I just did a test with the dropbox app, I created a target folder with a fake folder structure and the issue didn't happen.
